### ISSUE TITLE
RememberMe + Socialite

### ISSUE THREAD

**Issue by lsmith77** [laravel/framework#45945]
- Laravel Version: v9.50.1
- PHP Version: 8.0.23
- Database Driver & Version:

### Description:

The issue is a regression caused by: https://github.com/laravel/framework/pull/34020

### Steps To Reproduce:

Setup a Laravel app with Socialite (Socialstream in my case). Use social login exclusively to authenticate. As a result password will be empty on the user and therefore the user will be immediately logged out after they were authenticated via their remember me cookie due to the `! $passwordHash` check in `AuthenticateSession::handle()`
https://github.com/laravel/framework/blob/35be2599c9ac3d58bf1578895c2e85ea4848a0d7/src/Illuminate/Session/Middleware/AuthenticateSession.php#L46

A "fix" would be to set a default password on all users but this feels like a security issue waiting to happen.

**Comment by driesvints** [comment]
You can circumvent this by setting a random password when registering the user. I'm not sure it's wise to change anything here right now with the change being so old.

**Comment by lsmith77** [comment]
@joelbutcher if I open a PR on socialstream for this, do you prefer a doc tweak or a change to enforce a random password if remember me is enabled?

**Comment by joelbutcher** [comment]
@lsmith77 I agree with Dries on this, setting a random password per user and emailing this out to them would be the best mitigation of the sec issue. 

But, since you're exclusively wanting to use OAuth for authentication and remove email + password auth, I'd prefer this to be a documented "gotcha" rather than a sec related issue Socialstreams tries to solve for all developers as everyone could potentially want a different solution for this.

**Comment by lsmith77** [comment]
@joelbutcher ok .. on what repo should this be documented? framework, jetstream or socialstream?

**Comment by joelbutcher** [comment]
@lsmith77 Please send documentation updates to [joelbutcher/socialstream-docs](https://github.com/joelbutcher/socialstream-docs)
